.pre-commit-config.yaml

```
@@ -0,0 +1,34 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+  - repo: https://github.com/compilerla/conventional-pre-commit
+    rev: v3.4.0
+    hooks:
+      - id: conventional-pre-commit
+        stages:
+          - commit-msg
+        args: []
+  - repo: local
+    hooks:
+      - id: no-fixmes
+        name: check for fixmes
+        description: 'Detect lines containing uppercase fixme'
+        entry: '[F][I][X][M][E]'
```

Maybe q comment about this strange format

Also, this would be enough, no?

```suggestion
        entry: '[F]IXME' 
```

---

maybe I should just keep these two hooks out of that config - I added them to force myself to fix an issue by marking it with `// FIXME`, but preventing me from accidentally committing that marker. Maybe the better solution would be to allow committing such a marker, but add a blocker in the CI that refuses to pass when new FIXME or TODO items are added. That way one can commit a stub, mark it as to be fixed before merging, but discuss the interfaces around that code in the PR.